src/components/rocm_smi/rocs.c

```diff
@@ -1585,8 +1590,7 @@ handle_derived_events(const char *v_name, int32_t dev, int64_t v_variant, int64_
             return ROCS_EVENT_TYPE__DERIVED;
         }
 
-        int64_t i;
-        for (i = 0; i <= ROCS_PCI_BW_VARIANT__CURRENT; ++i) {
+        for (int64_t i = 0; i <= ROCS_PCI_BW_VARIANT__CURRENT; ++i) {
```

Comment:
I would avoid declaring the int declaration inside the for loop header. As it could cause issues with older compilers.